__tests__/session.test.js

```diff
@@ -24,14 +24,23 @@ describe('Session', () => {
     await shutdownApp(app, db);
   });
 
-  it('should return 302 and a cookie when using valid credentials', async () => {
-    const res = await app.inject({
+  it('should login and logout when using valid credentials', async () => {
+    let res = await app.inject({
       method: 'post',
       url: '/session',
       body: { email: user.email, password: 'test' },
     });
     expect(res.statusCode).toBe(302);
-    expect(res.headers['set-cookie']).toBeTruthy();
+    const cookie = res.headers['set-cookie'];
+    expect(typeof cookie).toBe('string');
+    res = await app.inject({
+      method: 'delete',
+      url: '/session',
+      headers: {
+        Cookie: cookie
+      }
+    });
+    expect(res.statusCode).toBe(302);
```

Comment:
We have two different controllers. Would be better to have separate tests. And it would be good to try log out again and expect error

Reply:
okay, any ideas how to test valid logout without logging first?

Reply:
so why have separate tests, if test for logout is almost the same as "test for login" + logout? let them be in one test?

Reply:
Test is a specifitcation of your code. One controller should have at least one test. If another test needs to use another controller it doesn't mean that this test covers two controllers

Reply:
there are a lot of tests types and am aiming at a pragmatic test suite, not on names :)
the reason is simple - test should fail if there is a bug in what it tests, right?
if we will split this test into two, and `signout` will be using `signin` mechanism, we will be in situation where  `signout` test will be failing because of `signin` bug! so, for example, if app cannot do login, but can logout, only `signin` test should fail, and in case you propose they will fail both, which is not correct

Reply:
In this case we should drop all session tests, because we use sign-in sign-out logic in the other tests, task tests  in particular. 
Separated tests for separate controllers give you confidence that your controller works. All the other tests that use this controller could be refactored in the future

Reply:
as i see, we dont use signin/signup in tests...
agree on controllers, but the idea behind that is to not mix unrelated things and make tests more atomic and isolated, and in this case there is no technical possibility to separate signin from signout, so treat these two as one is not a bad thing, because its most atomic and isolated way to test